tests/test_monkeypatch.py

```diff
@@ -156,3 +156,69 @@ def test_extended_moderation_admin_no_redirect_invalid_mr_to_copy(self):
         self.assertEqual(ContentExpiry.objects.count(), 1)
         # We should still be redirected
         self.assertEqual(response.status_code, 302)
+
+    def test_extended_moderation_admin_update_existing_compliance_number_record(self):
+        """
+        If the target of the copy already has a compliance number, the record should be updated, rather than recreated
+        """
+        # Create db data for additional content expiry!
+        self.content_expiry_secondary = factories.PollContentExpiryFactory(
+            expires=self.expires_secondary,
+            version__state=PUBLISHED
+        )
+        self.moderation_request2 = ModerationRequestFactory(
+            collection=self.collection,
+            version=self.content_expiry_secondary.version,
+        )
+        self.root2 = RootModerationRequestTreeNodeFactory(
+            moderation_request=self.moderation_request2
+        )
+        ChildModerationRequestTreeNodeFactory(
+            moderation_request=self.moderation_request1, parent=self.root1
+        )
+
+        # Check that compliance numbers are different before we hit the copy endpoint!
+        self.assertNotEqual(ContentExpiry.objects.first().compliance_number,
```

Comment:
This is more of a suggestion than anything, but could also verify only 2 records exist here, more for others reading than actual test functionality.

Reply:
Added another length check although it is being done after the copy endpoint is called

Reply:
Perfect, it's more to show it's the same before and after the copy action, for users.